platforms/ios/src/TangramMap/TGMarker.mm

```diff
@@ -51,99 +62,148 @@ - (instancetype)initWithMapView:(TGMapViewController *)mapView
         type = TGMarkerType::none;
         self.visible = YES;
         self.drawOrder = 0;
-        self.map = mapView;
+        [self map:mapView];
     }
 
     return self;
 }
 
-- (void)setStylingString:(NSString *)styling
+- (NSError *)createNSError
+{
+    NSMutableDictionary* userInfo = [[NSMutableDictionary alloc] init];
+    // TODO: add enum for object error
+    // [userInfo setObject: forKey:@"TGMarker"];
+
+    return [NSError errorWithDomain:@"TGMarker"
+                               code:0
+                           userInfo:userInfo];
+}
+
+- (NSError *)stylingString:(NSString *)styling
 {
     _stylingString = styling;
     _stylingPath = nil;
 
-    if (!tangramInstance || !identifier) { return; }
+    if (!tangramInstance || !identifier) { return nil; }
```

Comment:
Unsure, in a context where exception might be thrown out to the user this can become quite annoying, technically this is not an error _yet_ since you don't know whether the styling is valid or not here, but I was also wondering whether something should be thrown here, so I'm open to suggestions.